includes/admin/class-wc-rest-payments-orders-controller.php

```diff
@@ -138,7 +141,9 @@ public function capture_terminal_payment( WP_REST_Request $request ) {
 			}
 
 			// Do not process intents that can't be captured.
-			$intent                   = $this->api_client->get_intent( $intent_id );
+			$request = Get_Intent::create();
+			$request->set_intent_id( $intent_id );
```

Comment:
I'm starting to think that for update/get request, we could maybe include the ID in the constructor. Do you think that makes sense? This is pretty descrptive as well though, no strong opinions here.

Reply:
The reason why I choose this approach is that we have dependencies in abstract class constructor and create method is static, and it needs to be compatible with parent class. I was thinking to add constructor injection for this situation, but we used setters for filling data into request classes, so I decided to go with the same approach. If you think that constructor parameter is also a valid option, I will gladly refactor it. For me, it is 50-50 and that's why I chose setter since we already using it.